Controller/Adminhtml/Item/MassDelete.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace SomethingDigital\AuctionItem\Controller\Adminhtml\Item;
+
+use Magento\Backend\App\Action;
+use Magento\Backend\App\Action\Context;
+use Magento\Backend\Model\View\Result\Redirect;
+use Magento\Framework\App\Action\HttpPostActionInterface;
+use Magento\Framework\Controller\ResultFactory;
+use Magento\Framework\Exception\NotFoundException;
+use Magento\Ui\Component\MassAction\Filter;
+use SomethingDigital\AuctionItem\Model\AuctionItemRepository;
+use SomethingDigital\AuctionItem\Model\ResourceModel\AuctionItem\CollectionFactory;
+
+/**
+ * Custom Controller for AuctionItem MassDelete
+ *
+ * Class MassDelete
+ */
+class MassDelete extends Action implements HttpPostActionInterface
+{
+    /**
+     * Authorization level
+     */
+    const ADMIN_RESOURCE = 'SomethingDigital_AuctionItem::grid';
+
+    /**
+     * @var CollectionFactory
+     */
+    protected $collectionFactory;
+
+    /**
+     * @var AuctionItemRepository
+     */
+    private $auctionItemRepository;
+
+    /**
+     * @var Filter
+     */
+    protected $filter;
+
+    /**
+     * Constructor
+     *
+     * @param Context $context
+     * @param Filter $filter
+     * @param CollectionFactory $collectionFactory
+     * @param AuctionItemRepository $auctionItemRepository
+     */
+    public function __construct(
+        Context $context,
+        Filter $filter,
+        CollectionFactory $collectionFactory,
+        AuctionItemRepository $auctionItemRepository
+    ) {
+        $this->filter = $filter;
+        $this->collectionFactory = $collectionFactory;
+        $this->auctionItemRepository = $auctionItemRepository;
+        parent::__construct($context);
+    }
+
+    /**
+     * AuctionItem delete action
+     *
+     * @return Redirect
+     */
+    public function execute(): Redirect
+    {
+        if (!$this->getRequest()->isPost()) {
+            throw new NotFoundException(__('Page not found'));
```

Comment:
I'm a little confused about this part. What happens when the Exception is thrown? Is a NotFoundException the right exception to throw if the post body is missing?

Reply:
I followed this [guide](https://devdocs.magento.com/guides/v2.4/extension-dev-guide/admin-grid.html#6-backend-controllers) in the devdocs which has the same check. I think it's just to make sure that if you navigate to the url in the browser or do a get request it throws an exception since it's a post request.

Reply:
Ah makes sense. I think I was misreading the method name and somehow thought it was getting the post data from the request